File: android/app/src/main/kotlin/com/mottu/marvel/AppEventChannel.kt
package com.mottu.marvel

import android.content.Context
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.EventChannel

class AppEventChannel {

    fun setup(flutterEngine: FlutterEngine, context:Context){
        EventChannel(flutterEngine.dartExecutor.binaryMessenger, "com.mottu.marvel/network_connectivity").setStreamHandler(
            NetworkConnectivityHandler(context)
        )
    }
}

File: ios/Runner/AppEventChannel.swift
//
//  AppEventChannel.swift
//  Runner
//
//  Created by vini on 01/09/24.
//

import Foundation


class AppEventChannel {
    
    func setup(controller: FlutterViewController){
        let eventChannel = FlutterEventChannel(name: "com.mottu.marvel/network_connectivity", binaryMessenger: controller.binaryMessenger)
        eventChannel.setStreamHandler(NetworkConnectivityHandler())
    }
    
}
